library/src/rocblas_auxiliary.cpp

```diff
@@ -755,3 +755,29 @@ catch(...) // catch all exceptions
 {
     return rocblas_status_internal_error;
 }
+
+// Convert rocblas_status to string
+extern "C" const char* rocblas_status_to_string(rocblas_status status)
+{
+#define CASE(x) \
+    case x:     \
+        return #x
+    switch(status)
+    {
+        CASE(rocblas_status_success);
+        CASE(rocblas_status_invalid_handle);
+        CASE(rocblas_status_not_implemented);
+        CASE(rocblas_status_invalid_pointer);
+        CASE(rocblas_status_invalid_size);
+        CASE(rocblas_status_memory_error);
+        CASE(rocblas_status_internal_error);
+        CASE(rocblas_status_perf_degraded);
+        CASE(rocblas_status_size_query_mismatch);
+        CASE(rocblas_status_size_increased);
+        CASE(rocblas_status_size_unchanged);
+    }
+#undef CASE
+    // We don't use default: so that the compiler warns us if any valid enums are missing
+    // from our switch. If the value is not a valid rocblas_status, we return this string.
+    return "<undefined rocblas_status value>";
+}
```

Comment:
Should this be part of the external rocBLAS API and added to rocBLAS/library/include/rocblas_auxiliary.h ? something like:

ROCBLAS_EXPORT const char* rocblas_status_to_string(rocblas_status status);

Reply:
Yes I want to call it externally.

Reply:
> Should this be part of the external rocBLAS API and added to rocBLAS/library/include/rocblas_auxiliary.h ? something like:
> 
> ROCBLAS_EXPORT const char* rocblas_status_to_string(rocblas_status status);

It is in rocblas-functions.h, which is where exported functions are declared. Rocblas does not declare exported functions in other headers.

Reply:
Thanks, I see it in line 5432 of rocblas-functions.h